flutter-for-react-native/index.md

```diff
@@ -0,0 +1,2411 @@
+    onTap: () {
+      Navigator.of(context).pushNamed("/b");
+    },
+  ),
+  elevation: 20.0,
+),
+```
+
+The `Scaffold` widget also includes an `AppBar` widget that automatically displays an appropriate IconButton to show the `Drawer` when a Drawer is available in the `Scaffold`. The `Scaffold` automatically handles the edge-swipe gesture to show the `Drawer`.
+
+<!-- skip -->
+```Dart
+// Flutter
+@override
+Widget build(BuildContext context) {
+  return new Scaffold(
+    drawer: new Drawer(
+      child: new ListTile(
+        leading: new Icon(Icons.change_history),
+        title: new Text('Screen2'),
+        onTap: () {
+          Navigator.of(context).pushNamed("/b");
+        },
+      ),
+      elevation: 20.0,
+    ),
+    appBar: new AppBar(
+      title: new Text("Home"),
+    ),
+    body: new Container(),
+  );
+}
+```
+
+
+
+|Android|iOS|
+|:---:|:--:|
+|<img src="images/navigation_android.gif?raw=true" style="width:300px;" alt="Loading">|<img src="images/navigation_iOS.gif?raw=true" style="width:300px;" alt="Loading">|
+
+<br>
+
+## Gesture detection and touch event handling
+
+To listen for and respond to gestures, Flutter includes taps, drags, and
+scaling. The gesture system in Flutter has two separate layers. The first layer
+includes raw pointer events, which describe the location and movement of
+pointers, (such as touches, mice, and styli movements), across the screen. The
+second layer includes gestures, which describe semantic actions that consist of
+one or more pointer movements.
+
+### How do I add a click or press listeners to a widget?
+
+In React Native, listeners are added to components using `PanResponder` or the `Touchable` components.
+
+<!-- skip -->
+```JavaScript
+// React Native
+<TouchableOpacity
+  onPress={() => {
+    console.log("Press");
+  }}
+  onLongPress={() => {
+    console.log("Long Press");
+  }}
+>
+  <Text>Tap or Long Press</Text>
+</TouchableOpacity>
+```
+
+For more complex gestures and combining several touches into a single gesture,
+ [`PanResponder`](https://facebook.github.io/react-native/docs/panresponder.html) is used.
+
+<!-- skip -->
+```JavaScript
+// React Native
+class App extends Component {
+
+  componentWillMount() {
+    this._panResponder = PanResponder.create({
+      onMoveShouldSetPanResponder: (event, gestureState) =>
+        !!getDirection(gestureState),
+      onPanResponderMove: (event, gestureState) => true,
+      onPanResponderRelease: (event, gestureState) => {
+        const drag = getDirection(gestureState);
+      },
+      onPanResponderTerminationRequest: (event, gestureState) => true
+    });
+  }
+
+  render() {
+    return (
+      <View style={styles.container} {...this._panResponder.panHandlers}>
+        <View style={styles.center}>
+          <Text>Swipe Horizontally or Vertically</Text>
+        </View>
+      </View>
+    );
+  }
+}
+```
+
+In Flutter, to add a click or press listeners to a widget, you can use buttons
```

Comment:
=> In Flutter, to add a click (or press) listener to a widget, use a button or a touchable widget that has an `onPress:` field. Or, add gesture detection to any widget by wrapping it in a [`GestureDetector`]().

Reply:
Done